### PR TITLE
V18.0.0.0 rc

### PR DESCRIPTION
**Pull request by lindsleycj**
PR for Cumulus v18.0.0 support.

### CODE REVIEW

**Review by mikedorfman — APPROVED** [review]
Looks reasonable to me!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,19 @@
 # CHANGELOG
 
+## v18.0.0.0
+
+* Upgrade to [Cumulus v18.0.0](https://github.com/nasa/cumulus/releases/tag/v18.0.0)
+* **NOTE** If you are upgrading from a version of CIRRUS earlier than `v17.0.0.3` you
+may observe a CloudFormation error.  Use the information in the `v17.0.0.3` entry below
+to resolve it
+
```
Comment (reweeden) [inline CHANGELOG.md:9]: Should we mention in the release notes that the terraform version is changing to `1.5.3`?

**Review by reweeden — APPROVED** [review]
At ASF we also ran into some different issues with the TEA CloudFormation update, but they were inconsistent across stacks. We didn't have any cloudformation failures, but we did have issues in prod with the API gateway stage/deployment which we were able to fix by changing the name of the API Gateway Deployment resource in the template and updating the stack with the new template to force CloudFormation to recreate the API Gateway Deployment. In our UAT account however, the update worked without issue so it seems to be a bit random, probably some sort of race condition with the order in which CloudFormation updates the Deployment and Stage.

File: CHANGELOG.md (hunk 1 above)
Comment (lindsleycj) [reply CHANGELOG.md:9]: @reweeden Done!

File: cumulus/.terraform.lock.hcl
Comment (lindsleycj) [inline cumulus/.terraform.lock.hcl:1]: @mikedorfman @reweeden @mckadesorensen Please note that I have included this lock file plus the same in data-persistence and tf.  This is a new feature of the later Terraform.  That locks the terraform provider versions.  I had to modify the .gitignore file slightly to allow it to be included.  Do you all have any concerns about it?

**Comment by lindsleycj** [comment]
One more thing.  To support python 3.10 mentioned in the [v18 release notes](https://github.com/nasa/cumulus/releases/tag/v18.0.0), we'll have to update the base image of our docker file.  I haven't tackled that yet.  Do you think I need to before we release this?

**Comment by lindsleycj** [comment]
@mattp0 I don't personally have any requirement for python3.10.  The 3.8 stuff you did uses the `amazon-linux-extras`.  It doesn't have python3.10.  

I also see that the `amazonlinux:2023` base image comes standard with python3.9 and has an option to install python3.11, but also no 3.10.

so if we really want python3.10 for anything we might have to use some other base image.  Am I seeing that correctly?

**Comment by mattp0** [comment]
> @mattp0 I don't personally have any requirement for python3.10. The 3.8 stuff you did uses the `amazon-linux-extras`. It doesn't have python3.10.
> 
> I also see that the `amazonlinux:2023` base image comes standard with python3.9 and has an option to install python3.11, but also no 3.10.
> 
> so if we really want python3.10 for anything we might have to use some other base image. Am I seeing that correctly?
> 
> Maybe we just skip it until someone has a need.

I think that is a good idea. We do need to think about pushing the amazon linux label but that is likely a larger effort, as it will have downstream effects for the users of cirrus. with 3.7 at EoL and 3.8 headed that way, we should probably look at making this upgrade in the future. https://devguide.python.org/versions/

File: cumulus/.terraform.lock.hcl
Comment (mattp0) [reply cumulus/.terraform.lock.hcl:1]: This has come up as an issue #179 for OBDAAC

**Pull request merged**